src/xrCore/XML/XMLDocument.cpp

```diff
@@ -5,23 +5,79 @@
 pcstr UI_PATH = UI_PATH_DEFAULT;
 pcstr UI_PATH_WITH_DELIMITER = UI_PATH_DEFAULT_WITH_DELIMITER;
 
-XMLDocument::XMLDocument() : m_xml_file_name(), m_root(nullptr), m_pLocalRoot(nullptr) {}
+XMLDocument::XMLDocument() : m_xml_file_name(), m_root(nullptr), m_pLocalRoot(nullptr), m_bIgnoreMissingEndTagError(false) {}
 
 XMLDocument::~XMLDocument() { ClearInternal(); }
 
 void XMLDocument::ClearInternal() { m_Doc.Clear(); }
 
-void ParseFile(pcstr path, CMemoryWriter& W, IReader* F, XMLDocument* xml)
+enum class ParseIncludeResult
 {
-    string4096 str;
+    Success,   /// There is a valid #include and 'out_include_name' returns the filename
+    Error,     /// There is a #include but there is some problem
+    NoInclude, /// We #include on this line
+};
 
+// Give a string of the form: '#include "filename"' we return the filename in 'out_include_name'
+ParseIncludeResult ParseInclude(pstr string, pcstr* out_include_name)
```

Comment:
Why do we use pointer here?
```suggestion
ParseIncludeResult ParseInclude(pstr string, pcstr& out_include_name)
```

Reply:
Yes good catch using a reference here is much cleaner